app/services/ipc.js

```diff
@@ -0,0 +1,30 @@
+import Ember from 'ember';
+import electronUtils from 'wsd-core/utils/electron';
+
+export default Ember.Service.extend(Ember.Evented, {
+    ipcRenderer: null,
+
+    setup() {
```

Comment:
Would it make sense to move all the checks performed when you call `on` to the `init` function instead? It might result in a simpler execution flow (you could avoid storing the property `ipcRenderer` and you could probably also remove the overload of `on` function).